src/test_workflow/integ_test/integ_test_suite_opensearch.py

```diff
@@ -121,16 +121,16 @@ def custom_node_endpoint_encoder(node_endpoint: NodeEndpoint) -> dict:
             self.repo_work_dir = os.path.join(
                 self.repo.dir, self.test_config.working_directory) if self.test_config.working_directory is not None else self.repo.dir
             (status, stdout, stderr) = execute(cmd, self.repo_work_dir, True, False)
-            self.test_result_data.append(
-                TestResultData(
-                    self.component.name,
-                    test_config,
-                    status,
-                    stdout,
-                    stderr,
-                    self.test_artifact_files
-                )
+            test_result_data = TestResultData(
```

Comment:
Nit: name this local var as `test_result_data_obj` or `test_result_data_local` to distinguish from `self.test_result_data`.